jujugui/static/gui/src/app/store/env/go.js

```diff
@@ -2264,6 +2283,184 @@ YUI.add('juju-env-go', function(Y) {
         response.changeSet = data.Response.Changes;
       }
       userCallback(response);
+    },
+
+    // EnvironmentManager facade API endpoints.
+    // XXX frankban: facade management should be improved. We should store
+    // the list of facades and their corresponding available versions upon
+    // login, and then activate/deactivate the ability to perform specific
+    // API calls based on the server capabilities. Also, facades could help
+    // namespacing client functions, so that we don't store all the API calls
+    // inside a common namespace, resulting in a more organized code. Perhaps
+    // this can be done as part of creating an external JavaScript API library.
+    environmentManagerFacadeVersion: 1,
+
+    /**
+      Create an environment within this system, using the given name.
+
+      @method createEnv
+      @param {String} envName The name of the new environment.
+      @param {String} userTag The name of the new environment owner, including
+        the "user-" prefix.
+      @param {Function} callback A callable that must be called once the
+        operation is performed. It will receive an object with an "err"
+        attribute containing a string describing the problem (if an error
+        occurred), or with the following attributes if everything went well:
+        - name: the name of the new environment;
+        - owner: the environment owner tag;
+        - uuid: the unique identifier of the new environment.
+      @return {undefined} Sends a message to the server only.
+    */
+    createEnv: function(envName, userTag, callback) {
+      var self = this;
+      var intermediateCallback;
+      if (callback) {
+        // Capture the callback. No context is passed.
+        intermediateCallback = Y.bind(self._handleCreateEnv, null, callback);
```

Comment:
In an effort to use less YUI

`intermediateCallback = self._handleCreateEnv.bind(null, callback);`